File: Examples/Getting-Started/Get-Configuration-Revision/nodeJS/get-configuration-revision.mjs
import "dotenv/config";

import {
  NetLDClient,
  NetLDError,
  decodeRevisionContent,
} from "./netld-example-client.mjs";

const network = process.env.NETLD_NETWORK || "Default";
const deviceIp = process.env.NETLD_DEVICE_IP || "10.95.1.40";
const configPath = process.env.NETLD_CONFIG_PATH || "/running-config";

const client = NetLDClient.fromEnv();
await client.login();

const pageData = await client.getConfigurationHistory({
  networks: [network],
  scheme: "ipAddress",
  data: deviceIp,
});

const matches = (pageData?.configHistoryItems || []).filter(
  (item) => item.path === configPath,
);
if (matches.length === 0) {
  throw new NetLDError(
    `No configuration history item for "${configPath}" was found on ${deviceIp}@${network}.`,
  );
}

const historyItem = matches[0];
const revision = await client.retrieveRevision({
  network: historyItem.managedNetwork,
  ipAddress: historyItem.ipAddress,
  configPath: historyItem.path,
  timestamp: historyItem.lastChanged,
});
if (!revision) {
  throw new NetLDError("Configuration.retrieveRevision returned no revision.");
}

const { content, ...metadata } = revision;
console.log("Revision metadata:");
console.log(JSON.stringify(metadata, null, 2));

const decodedContent = decodeRevisionContent(revision);
if (decodedContent === null) {
  console.log("Revision content is empty or binary; Base64 content was not printed.");
} else {
  console.log("Decoded revision content:");
  console.log(decodedContent);
}

File: Examples/Getting-Started/Get-Configuration-Revision/Python/get_configuration_revision.py
import json
import os

from netld_example_client import NetLDClient, NetLDError, decode_revision_content


network = os.environ.get("NETLD_NETWORK", "Default")
device_ip = os.environ.get("NETLD_DEVICE_IP", "10.95.1.40")
config_path = os.environ.get("NETLD_CONFIG_PATH", "/running-config")

client = NetLDClient.from_env()
client.login()

page_data = client.get_configuration_history(
    networks=[network],
    scheme="ipAddress",
    data=device_ip,
)

matches = [
    item
    for item in (page_data or {}).get("configHistoryItems", [])
    if item.get("path") == config_path
]
if not matches:
    raise NetLDError(
        f'No configuration history item for "{config_path}" was found on {device_ip}@{network}.'
    )

history_item = matches[0]
revision = client.retrieve_revision(
    network=history_item["managedNetwork"],
    ip_address=history_item["ipAddress"],
    config_path=history_item["path"],
    timestamp=history_item["lastChanged"],
)
if revision is None:
    raise NetLDError("Configuration.retrieveRevision returned no revision.")

metadata = {key: value for key, value in revision.items() if key != "content"}
print("Revision metadata:")
print(json.dumps(metadata, indent=2))

decoded_content = decode_revision_content(revision)
if decoded_content is None:
    print("Revision content is empty or binary; Base64 content was not printed.")
else:
    print("Decoded revision content:")
    print(decoded_content)
